samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -905,14 +908,171 @@ def _generate_makefile_rule_for_lambda_resource(
+    terraform_built_artifacts_script_path = Path(output_dir, TERRAFORM_BUILD_SCRIPT).relative_to(
+        terraform_application_dir
+    )
+    return show_command_template.format(
+        python_command_name=python_command_name,
+        terraform_built_artifacts_script_path=terraform_built_artifacts_script_path,
+        jpath_string=jpath_string,
+        project_root_dir=terraform_application_dir,
+    )
+
+
+def _build_jpath_string(sam_metadata_resource: SamMetadataResource, resource_address: str) -> str:
+    """
+    Constructs the JPath string for a given sam metadata resource from the planned_values
+    to the build_output_path as is created by the Terraform plan output
+
+    Parameters
+    ----------
+    sam_metadata_resource: SamMetadataResource
+       Full address of a Terraform resource
+
+    resource_address: str
+        Full address of a Terraform resource
+
+    Returns
+    -------
+    str
+       Full JPath string for a resource from planned_values to build_output_path
+    """
+    jpath_string_template = (
+        "|planned_values|root_module{child_modules}|resources"
+        '[?address=="{resource_address}"]|values|triggers|built_output_path'
+    )
+    child_modules_template = "|child_modules[?address=={module_address}]"
+    module_address = sam_metadata_resource.current_module_address
+    full_module_path = ""
+    if module_address:
+        parent_modules = _get_parent_modules(module_address)
+        if parent_modules:
+            for module in parent_modules:
+                full_module_path += child_modules_template.format(module_address=module)
+    return jpath_string_template.format(child_modules=full_module_path, resource_address=resource_address)
+
+
+def _get_parent_modules(module_address: str) -> Optional[List[str]]:
+    """
+    Convert an a full Terraform resource address to a list of module
+    addresses from the root module to the current module
+
+    e.g. "module.level1_lambda.module.level2_lambda" as input will return
+    ["module.level1_lambda", "module.level1_lambda.module.level2_lambda"]
+
+    Parameters
+    ----------
+    module_address: str
+       Full address of the Terraform module
+
+    Returns
+    -------
+    Optional[List[str]]
+       List of module addresses starting from the root module to the current module
+    """
+    if not module_address:
+        return None
+
+    # Split the address on "." then combine it back with the "module" prefix for each module name
+    modules = module_address.split(".")
+    modules = [".".join(modules[i : i + 2]) for i in range(0, len(modules), 2)]
+
+    # Prefix each nested module name with the previous
+    previous_module = modules[0]
+    full_path_modules = [previous_module]
+    for module in modules[1:]:
+        module = previous_module + "." + module
+        previous_module = module
+        full_path_modules.append(module)
+    return full_path_modules
```

Comment:
Possibly very slightly cleaner alternative. Not necessary to incorporate this, up to you.

```
    if not module_address:
        return []

    module_separator = "module."
    modules = module_address.split(module_separator)
    # remove empty strings
    modules = list(filter(None, modules))
    
    full_path_modules = [module_separator + modules[0]]
    for index, module in enumerate(modules[1:]):
        # adds previous path and current module, in a sort of accumulator
        curr_path = full_path_modules[index] + module_separator + module
        full_path_modules.append(curr_path)
        
    return full_path_modules
```

Reply:
I tried using this solution but it didn't work out of the box since any modules which aren't the last one will be suffixed with a `.` when splitting, so we need to remove it from all of those module paths. I will leave it as is for now unless there are any strong opinions.

Reply:
hmmm not exactly sure what you mean. I tested this with a couple examples and it seemed to work.

Reply:
but our solutions are not very different, so again not a huge deal.

However, maybe we should check `modules` is not empty, before doing `modules[0]`